### PR TITLE
Add support to list users per organization, project and update the system-wide listing

### PR DESCRIPTION
**Pull request by MarkusH**
## Summary of the changes / Why this is an improvement

- Add `croud organizations users list`
- Add `croud projects users list`
- Update `croud users list` to show user roles and allow filtering for users w/o any roles.

## Checklist

 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: croud/cmd.py [hunk 1]
```diff
@@ -39,6 +39,17 @@ def project_name_arg(req_args: _ArgumentGroup, opt_args: _ArgumentGroup) -> None
     )
 
 
+def no_roles_arg(req_args: _ArgumentGroup, opt_args: _ArgumentGroup) -> None:
+    opt_args.add_argument(
+        "--no-roles",
+        nargs="?",
```
Comment (chaudum) [inline croud/cmd.py]: IMO an argument with `action="store_true"` is a better fit for this type of argument.

File: croud/__main__.py [hunk 2]
```diff
@@ -410,7 +437,12 @@
             "list": {
                 "help": "List all users within the specified organization.",
```
Comment (chaudum) [inline croud/__main__.py]: I think we should update the help text of the command.

File: croud/organizations/users/commands.py [hunk 3]
```diff
@@ -39,6 +39,20 @@ def org_users_add(args: Namespace):
...
+    """
+    Lists organization users
+    """
+
+    client = Client.from_args(args)
+    client.send(RequestMethod.GET, f"/api/v2/organizations/{args.org_id}/users/")
+    if client._data and client._output_fmt in {"table"}:
+        for user in client._data:
+            user["role_fqn"] = user["organization_roles"][0]["role_fqn"]
+    client.print(keys=["uid", "email", "username", "role_fqn"])
```
Comment (chaudum) [inline croud/organizations/users/commands.py]: What do you think about changing the signature of `client.print()` to also allow a list of "transformer" functions for the specified keys:

```suggestion
    client.print(
        keys=["uid", "email", "username", "role_fqn"],
        transform=[identity, identity, identity, transform_org_role],
    )
    
def identity(key, item):
    return item[key]
    
def transform_org_role(key, item):
    return item["organization_roles"][0][key]
```

Comment (chaudum) [reply croud/organizations/users/commands.py]: Same applies for the other transformations.

Comment (MarkusH) [reply croud/organizations/users/commands.py]: I like the idea.

**Comment by MarkusH** [comment]
I added the transforms and updated the docs.

**Comment by MarkusH** [comment]
> 1st and 2nd commit need to be squashed because the 1st commit requires code of the 2nd.

Nope, that's an illusion due to GitHub's temporal sorting of commits, and not hierarchical sorting. Running `tox -e py37` after each commit works perfectly fine.

**Pull request merged**